scale/codec.go

```diff
@@ -197,20 +208,12 @@ func (pe Encoder) Encode(value interface{}) error {
 		}
 
 	case reflect.Struct:
-		encodeable := reflect.TypeOf((*Encodeable)(nil)).Elem()
-		if t.Implements(encodeable) {
```

Comment:
I wonder why this existing code doesn't work for you ?

Reply:
It did work, I just moved the check for encodeable to the top. Before, you could not specify custom encoding for non-structs.